src/Microsoft.Health.Fhir.TemplateManagement/Overlay/OverlayFileSystem.cs

```diff
@@ -0,0 +1,143 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Health.Fhir.TemplateManagement.Models;
+
+namespace Microsoft.Health.Fhir.TemplateManagement.Overlay
+{
+    public class OverlayFileSystem : IOverlayFileSystem
+    {
+
+        private readonly string _workingFolder;
+        private readonly string _workingImageLayerFolder;
+        private readonly string _workingBaseLayerFolder;
+
+        public OverlayFileSystem(string workingFolder)
+        {
+            _workingFolder = workingFolder;
+            _workingImageLayerFolder = Path.Combine(workingFolder, Constants.HiddenLayersFolder);
+            _workingBaseLayerFolder = Path.Combine(workingFolder, Constants.HiddenBaseLayerFolder);
+        }
+
+        public OCIFileLayer ReadMergedOCIFileLayer()
+        {
+            var filePaths = Directory.EnumerateFiles(_workingFolder, "*.*", SearchOption.AllDirectories);
+            Dictionary<string, byte[]> fileContent = new Dictionary<string, byte[]>() { };
+            foreach (var oneFile in filePaths)
+            {
+                if (!Path.GetRelativePath(_workingFolder, oneFile).Contains(Constants.HiddenImageFolder))
+                {
+                    fileContent.Add(Path.GetRelativePath(_workingFolder, oneFile), File.ReadAllBytes(oneFile));
+                }
+            }
+
+            OCIFileLayer fileLayer = new OCIFileLayer() { FileContent = fileContent };
+            return fileLayer;
+        }
+
+        public void WriteMergedOCIFileLayer(OCIFileLayer oneLayer)
```

Comment:
>WriteMergedOCIFileLayer [](start = 20, length = 23)

add override option

Reply:
Could we only add override option in command line tool?